src/Fcf/Utils.hs

```diff
@@ -58,3 +65,59 @@ class IsBool (b :: Bool) where
 
 instance IsBool 'True  where _If a _ = a
 instance IsBool 'False where _If _ b = b
+
+-- * Case splitting
+
+infix 0 -->
+
+data Match j k
+  = Match_ j k
+  | Is_ (j -> Exp Bool) k
+  | Any_ k
+  | Else_ (j -> Exp k)
+
+-- | Equivalent (limited) of @\\case { .. }@ syntax at value level. Supports
```

Comment:
"(Limited) equivalent"? I think you can also leave off "at value level", and replace "type" with "value" in the next sentence (to me types are only things of kind `Type`, or sometimes `... -> Type`).